webui/react/src/components/TaskTable.tsx

```diff
@@ -1,40 +1,65 @@
-import { Table } from 'antd';
+import { Table, Tooltip } from 'antd';
 import { ColumnsType, ColumnType } from 'antd/lib/table';
 import React, { MouseEventHandler } from 'react';
 
-import { actionsColumn, Renderer, startTimeColumn, stateColumn,
-  userColumn } from 'table/columns';
 import { AnyTask, CommandTask, CommandType, CommonProps } from 'types';
 import { alphanumericSorter } from 'utils/data';
 import { canBeOpened } from 'utils/task';
 import { commandTypeToLabel } from 'utils/types';
 
+import Badge, { BadgeType } from './Badge';
 import Icon from './Icon';
 import { makeClickHandler } from './Link';
 import linkCss from './Link.module.scss';
-import css from './TaskTable.module.scss';
+import {
+  actionsColumn, ellipsisRenderer, Renderer, startTimeColumn, stateColumn, userColumn,
+} from './Table';
+import css from './Table.module.scss';
 
 interface Props extends CommonProps {
   tasks?: CommandTask[];
 }
 
-const typeRenderer: Renderer<CommandTask> = (_, record) =>
-  (<Icon name={record.type.toLowerCase()}
-    title={commandTypeToLabel[record.type as unknown as CommandType]} />);
+const idRenderer: Renderer<CommandTask> = id => {
+  const shortId = id.split('-')[0];
```

Comment:
non-blocking: This strikes me as a potentially unsafe operation. Will `id.split('-')` always return an array of at least one element? Do we want a fallback in case?

Reply:
good observation. TL;DR, Typescript and the behavior of `split` provides protection.

`id` is strongly typed to be a `string` by Typescript and with it guaranteed to be a string (even an empty string). Splitting on an empty string or splitting a string with a character that doesn't match will return an array with the original string as the single element.